### PR TITLE
[API Compat] [#4810] Enable developers to perform binary compatibility check on local machine

### PR DESCRIPTION
**Pull request by MartinLuccanera**
#minor
Addresses issue # 4810

## Description
Adds a script that enables devs to locally check binaries for compatibility issues via usage of [ApiCompat](https://github.com/dotnet/arcade/tree/main/src/Microsoft.DotNet.ApiCompat).

_ApiCompat version: 1.0.0-beta.21161.1_
_Packages version: 4.6.3_

> ApiCompat and package versions are being set to be the same ones used for the _botbuilder-dotnet-ci_ pipeline, but we'll probably have to evaluate in the future updating both versions to avoid missing some differences on the compared binaries.

## Specific Changes
  - Updated `botbuilder-dotnet-ci.yml`. Added variable to run apiCompat script only if building locally.
  - Added `ExecuteApiCompat.ps1` . Script that downloads and installs dependencies to check for compatibility issues.
  - Updated `README.md` with information related to local ApiCompat script.
  - Updated `Directory.Build.props`. Added ApiCompat version property to pass as parameter to the script.
  - Updated `.gitignore` to ignore ApiCompat generated folders.
  - Added ApiCompat script to run as a post-build event for all relevant projects.
  - Added `PackageId` property for the following projects where it was missing:
       - Microsoft.Bot.Builder.Azure.csproj
       - Microsoft.Bot.Builder.Dialogs.csproj
       - Microsoft.Bot.Builder.TemplateManager.csproj
       - Microsoft.Bot.Schema.csproj
       - Microsoft.Bot.Builder.Integration.AspNet.Core.csproj

### CODE REVIEW

File: build/ExecuteApiCompat.ps1 [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  to match API definitions between provided version and current build for provided project name.
+
+  .PARAMETER Path
+  Specifies the path to project root folder.
+
+  .PARAMETER Name
+  Specifies the name for the specific project we are building.
+
+  .PARAMETER Version
+  Specifies the version of the library to compare the implementation with..
```
Comment (ceciliaavila) [inline build/ExecuteApiCompat.ps1:16]:
```suggestion
  Specifies the version of the library to compare the implementation with.
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Let's change the title to something like `[API Compat][#4810] Enable developers to perform binary compatibility check on local machine`

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]
The Build locally passed with the latest changes and show no errors, nice!

**Review by santgr11 — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
The name of the branch is not okay, take into consideration that we always use the format southworks/(add,update,fix,remove,etc.)/feature

**Pull request closed without merging**